Userland/Libraries/LibCore/ConfigFile.cpp

```diff
@@ -52,16 +65,19 @@ ConfigFile::~ConfigFile()
 
 void ConfigFile::reparse()
 {
-    m_groups.clear();
+    if (!m_file->is_open()) {
+        if (m_fd.has_value() && !m_file->open(*m_fd, OpenMode::ReadOnly, File::ShouldCloseFileDescriptor::No))
+            return;
+        if (!m_fd.has_value() && !m_file->open(OpenMode::ReadOnly))
+            return;
+    }
```

Comment:
This feels a bit weird, can we keep the file always open instead of closing and reopening?

Reply:
The weirdness comes from the way `ConfigFile` is being used, I haven't found a better way myself.

Most users of `ConfigFile` don't change the content and just read from a file and as such haven't pledged `wpath`. If the content is changed `ConfigFile` will implicitly write or `sync` the data back to the file on destruction or by calling the `sync` function. In this case it not only needs write access but also uses `OpenMode::Truncate` to override old data.

My patch currently reproduces the old behavior of reopening the file if needed with `OpenMode::WriteOnly` and `OpenMode::Truncate` if no fd is specified, if a fd is specified it will keep the fd open and assume the code that opened the fd initially has set correct flags for the usage in `ConfigFile`.

Reply:
I've done some thinking, the result of that is to provide `Core::ConfigFile::AllowWrite` to `open(String const& filename)` so that if a user of the api requires to be able to write to the config the actual file is opened with the correct `OpenMode`.